ocaml/xapi/xapi_network.ml

```diff
@@ -162,31 +184,43 @@ let counter = ref 0
 let mutex = Mutex.create ()
 let stem = "xapi"
 
-let pool_introduce ~__context ~name_label ~name_description ~mTU ~other_config ~bridge =
+let pool_introduce ~__context ~name_label ~name_description ~mTU ~other_config ~bridge ~managed =
   let r = Ref.make() and uuid = Uuid.make_uuid() in
   Db.Network.create ~__context ~ref:r ~uuid:(Uuid.to_string uuid)
     ~current_operations:[] ~allowed_operations:[]
-    ~name_label ~name_description ~mTU ~bridge
+    ~name_label ~name_description ~mTU ~bridge ~managed
     ~other_config ~blobs:[] ~tags:[] ~default_locking_mode:`unlocked ~assigned_ips:[];
   r
+let choose_bridge_name ~__context ~bridge ~bridges =
+  if bridge = "" then
```

Comment:
Take the if-statement out of here (see below comment), and eliminate the `loop` function:
```
let rec choose_bridge_name bridges =
  let name = stem ^ (string_of_int !counter) in
  incr counter;
  if List.mem name bridges then
    choose_bridge_name bridges
  else
    name
```